traveler-ios-ui/TravelerKitUI/ViewControllers/AvailabilityViewController.swift

```diff
@@ -20,7 +20,7 @@ class AvailabilityViewController: UIViewController {
 
     weak var delegate: AvailabilityViewControllerDelegate?
 
-    var product: Product?
+    var product: BookingItemDetails?
```

Comment:
I shouldn't have to have the details to do this part should I?

Reply:
There's a caveat I didn't really knew how to solve in regards to this. This VC is supposed to fetch availabilities for a booking product item. Right now, the refactoring doesnt consider a bookingItem. The flow is as follows:

CatalogGroup -> Query Item/ Product Item -> BookingItemDetails / ParkingItemDetails / etc.. 

A specific booking item would be essentially a product item with its type set to bookable. I felt that this was too much repetition. Since this flow is triggered AFTER BookingItemDetails then I figured I might as well use that model.

We want the fetch call to be specific to booking and specific to that detail. If it remained a product item then the fetch call wouldn't be specific (ProductItems of parking type also conform to product).